### PR TITLE
Logging on Retry policy and Temporary add AZURE_LOG_LEVEL to 2 for investigating

### PR DESCRIPTION
**Pull request by mssfang**
Investigate this error:
```
com.azure.core.exception.HttpResponseException: Status code 429, "{"type":"https://azconfig.io/errors/too-many-requests","title":"The amount of requests sent to the server have surpassed the assigned quota","policy":"TotalRequests","status":429}"
```
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=119198&view=logs&j=3f0b3bd8-2d88-59b8-ec21-d9a54670d8fb

Add logging info to the RetryPolicy
and temporary turn on AZURE_LOG_LEVEL to 2 for investigation of above 429 too many request error

### CODE REVIEW

File: sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java [hunk 1]
```diff
@@ -54,6 +58,8 @@ public RetryPolicy(int maxRetries, Duration delayDuration) {
         return next.clone().process()
             .flatMap(httpResponse -> {
                 if (shouldRetry(httpResponse, tryCount)) {
+                    logger.info("[Retrying] Try count: {}, HTTP response: {}, Original HTTP Request: {}",
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java]: Let's move this into `determineDelayDuration` so we can also get the delay that will be used. Another option is setting a local variable to the delay duration and log it here.

Comment (conniey) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java]: In addition, I don't think it is necessary to log the entire HTTP response and HTTP request. There is an HTTP pipeline policy that does this already, the HttpDetailLogLevel policy.

Outputting the retry attempt and the delay should be sufficient.

**Comment by danieljurek** [comment]
The changes to `tests.yml` look OK... Is the goal here to investigate or permanently have the log level set to 2? 

If the goal is to investigate we can do investigation out of a branch and playground or use a draft PR.

**Comment by joshfree** [comment]
@mssfang I'm tagging this as `do not merge` so it doesn't accidentally ship in preview 4

**Comment by mssfang** [comment]
Will not merge this for preview 4 now. As we need to have a proper discussion about the logging in retry.

**Pull request merged**